### DISCUSSION TITLE
Get currently connected clients list

### DISCUSSION THREAD

**Discussion by The-Gk** [miguelgrinberg/Flask-SocketIO#1585] [Q&A]
Sir is there any way to get currently connected users lists of sid? And get users list of sid from a room?

**Comment by miguelgrinberg** [comment]
No. You can keep track of the users that are connected by adding connect and disconnect handlers.

**Comment by The-Gk** [comment]
The disconnect event isn't working or triggering on when a client closes connection . Can you please show a working example sir?

**Comment by miguelgrinberg** [comment]
There is an example in this repository.

**Comment by The-Gk** [comment]
Okay sir thanks. But I have a doubt. Can we make large big Web app for millions of incoming requests with flask and flask socket io? And does 
 flask can be used for making big Web apps?

**Comment by miguelgrinberg** [comment]
@Gokul-44 Do you have any specific concerns? Big apps are hard to write, it has nothing to do with Flask or any framework and everything to do with the developer.

**Comment by The-Gk** [comment]
Sir just think of a website having minimum 2k requests per second. Does flask can handle it? And if we use flask socket io we can use only one worker so can we make another python file and use threads or multiprocessing for running multiple servers on multiple ports using os module and add all the servers lists like http://localhost:5555, http://localhost:5556... in nginx and use load balancer like that?

 I don't know if you get what I meant sir but I made a website with flask and flask socket io and hosted in one single ec2 instance behind nginx. But because of using one worker it can't accept large concurrent requests. So can you suggest a way or can you tell me if the thing I mentioned above of using multiprocessing or threading work like load balancer?

**Comment by miguelgrinberg** [comment]
> Sir just think of a website having minimum 2k requests per second. Does flask can handle it?

As I said above, the request logic is going to be implemented by you. You have to write efficient code in your request handler that allows you to scale as much as you need. Flask has nothing to do with this, the amount of work that Flask does compared to your own business logic is in the noise.

> But because of using one worker

Have you read the documentation? You can use multiple workers. You can scale Socket.IO horizontally as much as you need.

**Comment by The-Gk** [comment]
Okay sir. One thing this I have only one ec2 server instance so can run multiple server with different ports parallel like
# main_app.py

import os, threading
def app1():
	os.system("gunicorn -w 1 --threads 100 app:app") #port 5000
def app2():
	os.system("gunicorn -w 1 --threads 100 app2:app") #port 5001
def app3():
	os.system("gunicorn -w 1 --threads 100 app3:app") #port 5002

# nginx conf
upstream socketio_nodes {
    ip_hash;

    server 127.0.0.1:5000;
    server 127.0.0.1:5001;
    server 127.0.0.1:5002;
    
}

server {
    listen 80;
    server_name example.com;

    location / {
        include proxy_params;
        proxy_pass http://127.0.0.1:5000;
    }

    location /static {
        alias <path-to-your-application>/static;
        expires 30d;
    }

    location /socket.io {
        include proxy_params;
        proxy_http_version 1.1;
        proxy_buffering off;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_pass http://socketio_nodes/socket.io;
    }
}

Can we run loadbalance like this by running main_app.py and nginx 8n same server instance? Sorry sir if iam wrong..

**Comment by miguelgrinberg** [comment]
It's really hard for me to look at code or nginx configuration and tell you if it is correct or incorrect. In general the approach is correct, as long as you also have a message queue, which is required when you use more than one server.

**Comment by The-Gk** [comment]
Okay sir. But I face a problem now the when I host with gunicorn or uwgi in a ec2 instance it is very slow for opening in browser. But if I use flask development server it opens very fast. Do you know what will be the reason sir?

**Comment by miguelgrinberg** [comment]
I don't, sorry. Have you looked at logs?
